zerver/migrations/0543_set_default_value_for_realm_direct_message_initiator_group_and_more.py

```diff
@@ -0,0 +1,63 @@
+# Generated by Django 4.2.8 on 2024-01-01 10:03
+
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import OuterRef
+
+
+def set_default_value_for_direct_message_initiator_group_and_more(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Realm = apps.get_model("zerver", "Realm")
+    NamedUserGroup = apps.get_model("zerver", "NamedUserGroup")
+
+    EVERYONE_GROUP_NAME = "role:everyone"
+    NOBODY_GROUP_NAME = "role:nobody"
+    PRIVATE_MESSAGE_POLICY_DISABLED = 2
+
+    Realm.objects.filter(
+        direct_message_initiator_group=None, private_message_policy=PRIVATE_MESSAGE_POLICY_DISABLED
+    ).update(
+        direct_message_initiator_group=NamedUserGroup.objects.filter(
+            name=NOBODY_GROUP_NAME, realm=OuterRef("id"), is_system_group=True
+        ).values("pk")
+    )
+    Realm.objects.filter(
+        direct_message_initiator_group=None,
+    ).exclude(private_message_policy=PRIVATE_MESSAGE_POLICY_DISABLED).update(
+        direct_message_initiator_group=NamedUserGroup.objects.filter(
+            name=EVERYONE_GROUP_NAME, realm=OuterRef("id"), is_system_group=True
+        ).values("pk")
+    )
```

Comment:
What's the logic behind this migration? I'd have expected us to set the initiator permission to `EVERYONE_GROUP_NAME` for every organization, since we previously had no restrictions on initiation, just on actually sending.

Reply:
Actually, this was mentioned in the [issue description](https://github.com/zulip/zulip/issues/24467#issue-1597518482). 

> Migration: Organizations that have disabled private messages should have the "Nobody" value set for who can start a direct message thread, and (more importantly) "Direct messages disabled" for who must be included in direct messages. Otherwise, use the default settings.

But I have removed `NOBODY_GROUP_NAME` for `direct_message_initiator_group` now.